docs/src/app/(public)/blog/page.tsx

```diff
@@ -0,0 +1,77 @@
+import Link from 'next/link';
+import { notFound } from 'next/navigation';
+import Heading from '../../../components/heading';
+import { reader } from '../../../utils/reader';
+import { parseAndFormatPublishedDate } from '../../../utils';
+import { ArrowRightIcon } from '../../../components/icons/arrow-right';
+
+export default async function Blog() {
+  const blogPosts = await reader.collections.blog.all();
+
+  if (!blogPosts) notFound();
+
+  // Sort posts by publishedOn date, and filter out draft posts
+  const sortedPublishedBlogPosts = blogPosts
+    .map(post => {
+      const { parsedDate, formattedDate } = parseAndFormatPublishedDate(
+        post.entry.publishedOn
+      );
+
+      return {
+        ...post,
+        parsedDate,
+        formattedDate,
+      };
+    })
+    .sort((a: { parsedDate: Date }, b: { parsedDate: Date }) =>
+      a.parsedDate < b.parsedDate ? 1 : -1
+    )
+    .filter((post: { entry: { draft: boolean } }) => !post.entry.draft);
```

Comment:
If you move this `filter` up one level (i.e. chain it onto the `map`, then chain the `sort` to the `filter`) you will have less posts to `sort`.

Alternatively, you could switch the `map` for a `reduce` and handle the filtering during the same loop (see suggestion below).

I doubt we'll get to a point where we have so many blog posts that a more efficient sorting algorithm than `.sort()` will be needed but I think it'd be nice to improve the efficiency a little, with either just moving the `sort` or doing the below.

Let me know what you think of this (the filtering happens in the return ternary):

```suggestion
 type BlogPost = (typeof blogPosts)[number] & {
    formattedDate: string;
    parsedDate: Date;
  };

  const sortedPublishedBlogPosts = blogPosts
    .reduce<BlogPost[]>((acc, post): BlogPost[] | [] => {
      const { parsedDate, formattedDate } = parseAndFormatPublishedDate(
        post.entry.publishedOn
      );

      return post.entry.draft
        ? acc
        : [
            ...acc,
            {
              ...post,
              parsedDate,
              formattedDate,
            },
          ];
    }, [])
    .sort((a, b) => (a.parsedDate < b.parsedDate ? 1 : -1));
```

Reply:
Good find to move the filter above the sort.

IMO `.map.filter.sort` is much more readable than a `reduce`. For reduce to be more optimised we likely have to have tens of thousands or hundreds of thousands of blog posts. 

Also in production this code is only run when the site is built, so I'd take readability over efficiency in this case.

Reply:
No problem, sounds good!

Reply:
Ahh thanks @ChrisLaneAU! Doing the filtering before the sorting definitely make sense, thanks for catching that 🙏🏻 

I had to brush up a little about the `reduce()` method to understand your suggestion so thanks for writing that up!! Forever learnin' 🙌🏻  Chatted w Barney in standup as well and I think I'll stick with map + filter cause it feels easier to read, but defo valuable to know that the reduce would be more performant in this case

Reply:
Out of curiosity about where the crossover point would be I ran some jsbench tests and `.map.filter` was always faster than `.reduce` by itself. I tested 3, 300, 3000, 30 000, 300 000.

The code comparison was (I adjusted the `length` for different size data):
```
Array.from({length: 100}).flatMap(emptyVal => ([{a:{b: false}}, {a:{b: true}}, {a:{b: false}}])).map(val => val).filter(val => val.a.b);

Array.from({length: 100}).flatMap(emptyVal => ([{a:{b: false}}, {a:{b: true}}, {a:{b: false}}])).reduce((acc, val) => val.a.b ? [...acc, val] : acc, []);
```

It's not exactly concrete research but it's interesting to see! So yeah, stick with `.map.filter.sort` here and take any advice I give with a pinch of salt 😅

Reply:
Oh okay very interesting indeed!!

Reply:
> Out of curiosity about where the crossover point would be I ran some jsbench tests and `.map.filter` was always faster than `.reduce` by itself. I tested 3, 300, 3000, 30 000, 300 000.

I love that you tested this rather than just accept my hand-wavy made-up stat 😂 

It is funny we can never tell how the JS engine is optimised under the hood for these situations. Good research!